Added high-level diagrams

ivanmilevtues opened the pull request:
This PR includes high-level mermaid diagrams representing the CoSpred codebase. You can view them here: https://github.com/CodeBoarding/GeneratedOnBoardings/blob/main/starfish/on_boarding.md

The goal is to help new contributors quickly understand the code. We're especially interested in whether these diagrams are useful for scientists or others who use code but aren't full-time engineers—like at Pfizer. What does your current onboarding look like, and could something like this fit in?

We generate the diagrams using static analysis and LLMs. Feedback is very welcome! We're also building a GitHub Action to auto-update the diagrams on each main/release merge.

Full disclosure: we're exploring turning this into a startup and are still in the early stages.

Commit:
Added high-level diagrams

ivanmilevtues commented:
Hey, a quick update on our side, this week we released our diagram generaiton engine as an open-source project! If you are interested to see more on how the generation works you can do so at https://github.com/CodeBoarding/CodeBoarding

shachafl commented:
Hey @ivanmilevtues,
Thank you for contributing.
I apologize for the delayed response. I am extremely busy this summer and didn't have the bandwidth to explore your tool.
At a first glance this looks interesting, especially the flow and the automatic diagram generation and update. But I did notice that there are some issues with generated links, where it either does not generate a link or the link does not lead to the source code.
Your most recent builds might solve those issues, can you rerun on our repo?

Regarding onboarding `starfish`, I think the current documentations are reasonable including the manually drawn diagrams (although not homogenous, so some are better than others). But your tool does a better job at giving a high-level summary, yet allows to look in more details on the various components.

@berl you have better perspective than me regarding onboarding new people, and the burden of documentation, what are your thoughts?

ivanmilevtues committed:
Merge branch 'spacetx:master' into master

ivanmilevtues committed:
Updated diagrams

ivanmilevtues committed:
Updated graphics

ivanmilevtues commented:
@shachafl you are absolutely right! We have added a lot of new things and updated our approach (also [open-source now](github.com/Codeboarding/CodeBoarding)).

I regenerated the graphs for you, you can see how they render here: https://github.com/CodeBoarding/GeneratedOnBoardings/blob/main/starfish/on_boarding.md

ivanmilevtues committed:
Updated links

ivanmilevtues committed:
chore(codeboarding): reset baseline and add workflows

Amp-Thread-ID: https://ampcode.com/threads/T-019ecb81-dc76-76cb-8bf3-22a366c9be41
Co-authored-by: Amp <amp@ampcode.com>

ivanmilevtues committed:
fix(codeboarding): use configured LLM secret

Amp-Thread-ID: https://ampcode.com/threads/T-019ecb81-dc76-76cb-8bf3-22a366c9be41
Co-authored-by: Amp <amp@ampcode.com>

codeboarding[bot] committed:
chore(codeboarding): sync architecture baseline

ivanmilevtues committed:
test(codeboarding): use action feature branch

Amp-Thread-ID: https://ampcode.com/threads/T-019ecb81-dc76-76cb-8bf3-22a366c9be41
Co-authored-by: Amp <amp@ampcode.com>

codeboarding[bot] committed:
chore(codeboarding): sync architecture baseline
